type_lens/type_view.py

```diff
@@ -63,11 +63,15 @@ def __eq__(self, other: object) -> bool:
 
     def __repr__(self) -> str:
         cls_name = self.__class__.__name__
+        return f"{cls_name}({self.repr_type})"
 
-        raw = self.raw
-        if isinstance(self.raw, type):
-            raw = raw.__name__  # type: ignore[attr-defined]
-        return f"{cls_name}({raw})"
+    @property
+    def repr_type(self) -> str:
+        ann = self.annotation
+        if isinstance(ann, type) and not self.origin:
+            ann = ann.__name__  # type: ignore[attr-defined]
+
+        return str(ann).replace("typing.", "")
```

Comment:
Wouldn't this potentially also catch things that come from user supplied `typing` modules?

Reply:
yea :(. I could remove that part.

It's just annoying because the repr across different python versions does/does not include the `typing.` prefix. And it seems like a bad idea to circumvent the `__repr__` itself. Although perhaps i ought to go look at the impl and see whether it's straightforward to traverse or something...

Reply:
> It's just annoying because the repr across different python versions does/does not include the `typing.` prefix. 

Oh. I wasn't aware of that. Hmm. How about you check if it's a member of stdlib `typing`?

Reply:
that'll only work at the top level. `list[int] | set[int]` would be `typing.Union[typing.List[int], typing.Set[int]`. As i said above, it might be not-that-bad to traverse the object graph based on `origin` and repr each item individually, or it might suck a lot :P

Alternatively, we just let it vary across python versions 🤷

Reply:
I'm not entirely certain this if the new impl is foolproof, but it wasn't as difficult as i had imagined (thanks to `TypeView` itself), and it ought to now be more consistent across python versions anyway